firmware/common2015/CMakeLists.txt

```diff
@@ -1,43 +1,82 @@
+
+# append all of the accessory library sources
+list(APPEND common2015_SRC ${MBED_ASSEC_LIBS_SRCS})
+
+# set the names of each directory and then set the names of the subdirectories within each one
+set(DRIVERS_ROOT_NAME   drivers)
+set(MODULES_ROOT_NAME   modules)
+set(UTILS_ROOT_NAME     utils)
+# subdirectories
+set(DRIVERS             cc1101 cc1201 cc1201/cfg software-spi)
+set(MODULES             CommModule CommLink Console)
+set(UTILS               assert fp16 logger numparser rtos-mgmt)
+
+# generate a list of directories that we need to include from the values set above
+set(common2015_INCLUDE_DIRS 
+    ${CMAKE_CURRENT_SOURCE_DIR}/${DRIVERS_ROOT_NAME} 
+    ${CMAKE_CURRENT_SOURCE_DIR}/${MODULES_ROOT_NAME} 
+    ${CMAKE_CURRENT_SOURCE_DIR}/${UTILS_ROOT_NAME}
 )
 
-# tell CMake that the obj files all come from the ExternalProject
-# otherwise it'll complain that the files can't be found
-foreach(mbed_obj ${MBED_OBJECTS})
+# append all of the common2015 subdirectories listed in the variables above
+foreach(driver_subdir ${DRIVERS})
+    list(APPEND common2015_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/${DRIVERS_ROOT_NAME}/${driver_subdir})
+endforeach()
+foreach(module_subdir ${MODULES})
+    list(APPEND common2015_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/${MODULES_ROOT_NAME}/${module_subdir})
+endforeach()
+foreach(util_subdir ${UTILS})
+    list(APPEND common2015_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/${UTILS_ROOT_NAME}/${util_subdir})
+endforeach()
+
+# append all of the accessory library include directories
+list(APPEND common2015_INCLUDE_DIRS ${MBED_ASSEC_LIBS_INCLUDES})
+
+# do the same thing for the accessory libraries 
+foreach(assec_lib_file ${MBED_ASSEC_LIBS_SRCS})
     add_custom_command(
-        OUTPUT ${mbed_obj}
-        DEPENDS mbed_libraries
-        COMMAND "")
+        OUTPUT      ${assec_lib_file}
+        DEPENDS     ${MBED_ASSEC_LIBS_DEPENDS}
+        COMMAND     ""
+    )
 endforeach()
 
 # build the 'common2015' library that contains all the MBED stuff needed for
 # both the base station and robot firmware
-add_library(common2015 ${common2015_SRC})
-add_dependencies(common2015 mbed_libraries)
+add_library(common2015 ${common2015_SRC} ${CMAKE_BINARY_DIR}/common/git_version.cpp)
+
+# depends on the mbed libraries and all of the accessory libraries
+add_dependencies(common2015 mbed_libraries ${MBED_ASSEC_LIBS_DEPENDS})
+
 target_link_libraries(common2015 ${MBED_OBJECTS})
 target_link_libraries(common2015 ${MBED_LIBS})
 
+target_include_directories(common2015 PUBLIC ${common2015_INCLUDE_DIRS})
+
 # only build firmware library if required
 set_target_properties(common2015 PROPERTIES EXCLUDE_FROM_ALL TRUE)
 
+# the official mbed test builds
+set(MBED_TEST_UNIT 1)
+add_custom_target(mbed-test
+    COMMAND sudo python2 ${MBED_LIBS_REPO_DIR}/workspace_tools/make.py --mcu=${MBED_TARGET} --tool=GCC_ARM
+        --disk=/media/jonathan/MBED --serial=/dev/mbed --verbose -p ${MBED_TEST_UNIT} --jobs=0 --baud=9600
```

Comment:
Its from [mbed's automated test tools](https://developer.mbed.org/teams/SDK-Development/wiki/Mbed-SDK-automated-test-suite). It's something that pretty cool, but I don't think we'd get much use putting time into it it considering people working for mbed will detect an sdk error and fix it before we'd notice most likely.